File: cos_sim_test.py

from time import clock
import cos_sim

size = 50000
print('Generating 2 vectors of size {}. Similarity should be: -1.0.'.format(size))
A = [-10 for x in range(size)]
B = [10 for x in range(size)]

repeat = 50
print('Calculating Cosine Similarity. Repeating {}x.'.format(repeat))
avg_runtime = 0
similarity = None
for i in range(repeat):
    start = clock()
    similarity = cos_sim.get_cosine_similarity(A,B)
    end = clock()
    avg_runtime += (end-start)
avg_runtime = round(avg_runtime / repeat, 4)
print('Done. Average runtime: {} s. Similarity: {}.'.format(avg_runtime,similarity))

File: cos_sim_test.pl
#!/usr/bin/perl

use strict;
use Time::HiRes;
use lib '.'; #Redundant if @IRC includes current dir
use cos_sim;

my $size = 50000;
my (@A, @B);
print("Generating 2 vectors of size $size. Similarity should be: -1.0'.\n");

# Generate random values in vectors
foreach(0..$size) {
    $A[$_] = -10;
    $B[$_] = 10;
}

my $repeat = 50;
print("Calculating Cosine Similarity. Repeating $repeat x.\n");
my $avg_runtime = 0;
my $similarity;
foreach(0..100) {
  my $begin = Time::HiRes::time();
  $similarity = &cos_sim::get_cos_sim(\@A, \@B);
  my $end = Time::HiRes::time();
  $avg_runtime += ($end-$begin);
}
$avg_runtime /= $repeat;
printf("Done. Average runtime: %.4f s. Similarity: %0f.\n", $avg_runtime, $similarity);
